### ISSUE TITLE
21.3.0: pytest is failing

### ISSUE THREAD

**Issue by kloczek** [hynek/structlog#370]
I'm trying to package your module as an rpm package. So I'm using the typical build, install and test cycle used on building packages from non-root account.
- `python3 -Bm build --no-isolation -w`
- install .whl file in </install/prefix>
- "pytest with PYTHONPATH pointing to sitearch and sitelib inside </install/prefix>

May I ask for help because few units are failing:
```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-structlog-21.3.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-structlog-21.3.0-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.8.12, pytest-6.2.5, py-1.10.0, pluggy-0.13.1
rootdir: /home/tkloczko/rpmbuild/BUILD/structlog-21.3.0, configfile: pyproject.toml, testpaths: tests
plugins: rerunfailures-9.1.1, cov-2.12.1, forked-1.3.0, xdist-2.3.0, flake8-1.0.7, shutil-1.7.0, virtualenv-1.7.0, trio-0.7.0, mock-3.6.1, timeout-2.0.1, anyio-3.3.1, tornado-0.8.1, asyncio-0.15.1, httpbin-1.0.0, hypothesis-6.14.6, subtests-0.5.0, flaky-3.7.0, localserver-0.5.1
collected 505 items

tests/test_base.py ....................                                                                                                                              [  3%]
tests/test_config.py ........................................                                                                                                        [ 11%]
tests/test_contextvars.py .............                                                                                                                              [ 14%]
tests/test_dev.py ..s................................................ssss                                                                                            [ 25%]
tests/test_frames.py ............                                                                                                                                    [ 27%]
tests/test_generic.py .........                                                                                                                                      [ 29%]
tests/test_log_levels.py .........                                                                                                                                   [ 31%]
tests/test_loggers.py ................................................................................                                                               [ 47%]
tests/test_processors.py ..............s........s...s...s...s...s...s...s...s...s...s...s...s................................                                        [ 66%]
tests/test_stdlib.py .F............................................................................................                                                  [ 85%]
tests/test_testing.py .................                                                                                                                              [ 88%]
tests/test_threadlocal.py ......s..................                                                                                                                  [ 93%]
tests/test_twisted.py ............................                                                                                                                   [ 99%]
tests/test_utils.py ...                                                                                                                                              [100%]

================================================================================= FAILURES =================================================================================
__________________________________________________________________ TestLoggerFactory.test_ignores_frames ___________________________________________________________________

self = <tests.test_stdlib.TestLoggerFactory object at 0x7fe4f5744c70>

    def test_ignores_frames(self):
        """
        The name guesser walks up the frames until it reaches a frame whose
        name is not from structlog or one of the configurable other names.
        """
>       assert (
            "__main__"
            == additional_frame(
                LoggerFactory(
                    ignore_frame_names=["tests.", "_pytest.", "pluggy"]
                )
            ).name
        )
E       AssertionError: assert '__main__' == 'flaky.flaky_pytest_plugin'
E         - flaky.flaky_pytest_plugin
E         + __main__

tests/test_stdlib.py:87: AssertionError
========================================================================= short test summary info ==========================================================================
SKIPPED [1] tests/test_dev.py:55: Must be running on Windows.
SKIPPED [1] tests/test_dev.py:475: Needs rich.
SKIPPED [1] tests/test_dev.py:481: Needs rich.
SKIPPED [1] tests/test_dev.py:499: Needs better-exceptions.
SKIPPED [1] tests/test_dev.py:509: Needs better-exceptions.
SKIPPED [1] tests/test_processors.py:180: simplejson is missing.
SKIPPED [12] tests/test_processors.py:269: Skipped
SKIPPED [1] tests/test_threadlocal.py:151: Don't mix threads and greenlets.
FAILED tests/test_stdlib.py::TestLoggerFactory::test_ignores_frames - AssertionError: assert '__main__' == 'flaky.flaky_pytest_plugin'
================================================================ 1 failed, 485 passed, 19 skipped in 2.24s =================================================================
```

**Comment by kloczek** [comment]
I've edited pytest output (I've copiet wrong content .. sorry).
pytest fails in only one unit.

**Comment by hynek** [comment]
Should be fixed by e1324aa6374b649abfbb7b9425f5a274712f9791

**Comment by kloczek** [comment]
Just tested that commit it fixes reported issue
```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-structlog-21.3.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-structlog-21.3.0-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.8.12, pytest-6.2.5, py-1.10.0, pluggy-0.13.1
rootdir: /home/tkloczko/rpmbuild/BUILD/structlog-21.3.0, configfile: pyproject.toml, testpaths: tests
plugins: shutil-1.7.0, virtualenv-1.7.0, mock-3.6.1, hypothesis-6.14.6, cov-2.12.1, anyio-3.3.1, asyncio-0.15.1
collected 505 items

tests/test_base.py ....................                                                                                                                              [  3%]
tests/test_config.py ........................................                                                                                                        [ 11%]
tests/test_contextvars.py .............                                                                                                                              [ 14%]
tests/test_dev.py ..s................................................ssss                                                                                            [ 25%]
tests/test_frames.py ............                                                                                                                                    [ 27%]
tests/test_generic.py .........                                                                                                                                      [ 29%]
tests/test_log_levels.py .........                                                                                                                                   [ 31%]
tests/test_loggers.py ................................................................................                                                               [ 47%]
tests/test_processors.py ..............s........s...s...s...s...s...s...s...s...s...s...s...s................................                                        [ 66%]
tests/test_stdlib.py ..............................................................................................                                                  [ 85%]
tests/test_testing.py .................                                                                                                                              [ 88%]
tests/test_threadlocal.py ............s............                                                                                                                  [ 93%]
tests/test_twisted.py ............................                                                                                                                   [ 99%]
tests/test_utils.py ...                                                                                                                                              [100%]

========================================================================= short test summary info ==========================================================================
SKIPPED [1] tests/test_dev.py:55: Must be running on Windows.
SKIPPED [1] tests/test_dev.py:475: Needs rich.
SKIPPED [1] tests/test_dev.py:481: Needs rich.
SKIPPED [1] tests/test_dev.py:499: Needs better-exceptions.
SKIPPED [1] tests/test_dev.py:509: Needs better-exceptions.
SKIPPED [1] tests/test_processors.py:180: simplejson is missing.
SKIPPED [12] tests/test_processors.py:269: Skipped
SKIPPED [1] tests/test_threadlocal.py:227: Needs greenlet.
===================================================================== 486 passed, 19 skipped in 2.05s ======================================================================
```
Thank you :)
